data/maps/FallarborTown_BattleTentBattleRoom/scripts.inc

```diff
@@ -96,66 +97,69 @@ FallarborTown_BattleTentBattleRoom_EventScript_2009B3:: @ 82009B3
 	multichoice 20, 6, MULTI_GO_ON_REST_RETIRE, 1
 	switch VAR_RESULT
-	case 0, FallarborTown_BattleTentBattleRoom_EventScript_200AD8
-	case 1, FallarborTown_BattleTentBattleRoom_EventScript_200A78
-	case 2, FallarborTown_BattleTentBattleRoom_EventScript_200AA6
+	case 0, FallarborTown_BattleTentBattleRoom_EventScript_ContinueChallenge
+	case 1, FallarborTown_BattleTentBattleRoom_EventScript_AskPauseChallenge
+	case 2, FallarborTown_BattleTentBattleRoom_EventScript_AskRetireChallenge
 
-FallarborTown_BattleTentBattleRoom_EventScript_200A78:: @ 8200A78
+FallarborTown_BattleTentBattleRoom_EventScript_AskPauseChallenge:: @ 8200A78
 	msgbox BattleFrontier_BattleArenaBattleRoom_Text_SaveAndShutDown, MSGBOX_YESNO
 	switch VAR_RESULT
-	case NO, FallarborTown_BattleTentBattleRoom_EventScript_200A2A
-	case YES, FallarborTown_BattleTentBattleRoom_EventScript_200B51
-	case MULTI_B_PRESSED, FallarborTown_BattleTentBattleRoom_EventScript_200A2A
+	case NO, FallarborTown_BattleTentBattleRoom_EventScript_AskContinueChallenge
+	case YES, FallarborTown_BattleTentBattleRoom_EventScript_PauseChallenge
+	case MULTI_B_PRESSED, FallarborTown_BattleTentBattleRoom_EventScript_AskContinueChallenge
 
-FallarborTown_BattleTentBattleRoom_EventScript_200AA6:: @ 8200AA6
+FallarborTown_BattleTentBattleRoom_EventScript_AskRetireChallenge:: @ 8200AA6
 	message BattleFrontier_BattleArenaBattleRoom_Text_RetireFromChallenge
 	waitmessage
 	multichoicedefault 20, 8, MULTI_YESNO, 1, 0
 	switch VAR_RESULT
-	case 1, FallarborTown_BattleTentBattleRoom_EventScript_200A2A
-	case 0, FallarborTown_BattleTentBattleRoom_EventScript_20097E
-	case MULTI_B_PRESSED, FallarborTown_BattleTentBattleRoom_EventScript_200A2A
+	case 1, FallarborTown_BattleTentBattleRoom_EventScript_AskContinueChallenge
+	case 0, FallarborTown_BattleTentBattleRoom_EventScript_WarpToLobbyLost
+	case MULTI_B_PRESSED, FallarborTown_BattleTentBattleRoom_EventScript_AskContinueChallenge
 
-FallarborTown_BattleTentBattleRoom_EventScript_200AD8:: @ 8200AD8
+FallarborTown_BattleTentBattleRoom_EventScript_ContinueChallenge:: @ 8200AD8
 	closemessage
-	applymovement 2, FallarborTown_BattleTentBattleRoom_Movement_200BAB
+	applymovement 2, FallarborTown_BattleTentBattleRoom_Movement_AttendantReturnToPos
 	waitmovement 0
-	applymovement 1, FallarborTown_BattleTentBattleRoom_Movement_200B92
+	applymovement 1, FallarborTown_BattleTentBattleRoom_Movement_PlayerFaceBattle
 	waitmovement 0
-	goto FallarborTown_BattleTentBattleRoom_EventScript_20090F
+	goto FallarborTown_BattleTentBattleRoom_EventScript_NextOpponentEnter
 	waitstate
 
-FallarborTown_BattleTentBattleRoom_EventScript_200AF3:: @ 8200AF3
+FallarborTown_BattleTentBattleRoom_EventScript_WarpToLobbyWon:: @ 8200AF3
 	delay 60
 	frontier_set FRONTIER_DATA_CHALLENGE_STATUS, CHALLENGE_STATUS_WON
 	special LoadPlayerParty
 	warp MAP_FALLARBOR_TOWN_BATTLE_TENT_LOBBY, 255, 6, 6
 	waitstate
+
+@ Unreachable code block? The flow into the next block also doesnt make sense
```

Comment:
Why doesn't this make sense? It looks like it continues the script after warping into the lobby map.

Reply:
If it continues into the next block then every time the win streak is incremented there (which would apparently be occurring while the player is warping back the lobby) it would ask if the player is ready for the 2nd opponent. At that point they've already defeated all 3 opponents

Reply:
It looks like it was leftover porting this from Arena. It matches a block there, and I think these are the only direct uses of Arena functions in Fallarbor Tent.